In [1]:
# 导入包
import numpy as np
import pandas as pd
import os

In [2]:
pd.set_option('display.max_rows', None) # 显示所有行
pd.set_option('display.max_columns', None) # 显示所有列

In [3]:
# 文件根路径
filepath_cn = "./疫情数据/国内数据/"
filepath_fg = "./疫情数据/外国数据/"

# 国内数据分析及处理

## 分析处理国内历史每日数据

In [4]:
# 查看国内历史每日数据
china_day_data = pd.read_csv(filepath_cn+"国内历史每日数据.csv") # 这其实是每日累计数据
# 为了更直观看，将列名全部改为中文
china_day_data.columns = ['累计确诊', '疑似', '累计死亡', '累计治愈', '现有确诊', '现有重症患者', '境外输入', '死亡率', '治愈率', '日期', '无症状感染者']
# 日期列调到第一列
data_col = china_day_data.pop('日期') # 先弹出该列
china_day_data.insert(0, '日期', data_col) # 再将该列添加到第一列即可
china_day_data

日期   累计确诊     疑似  累计死亡   累计治愈   现有确诊  现有重症患者  境外输入  死亡率   治愈率  无症状感染者
0    1.16     45      0     2      8      0       0     0  4.4  17.8       0
1    1.17     62      0     2     12      0       0     0  3.2  19.4       0
2    1.18    198      0     3     17      0       0     0  1.5   8.6       0
3    1.19    275      0     4     18      0       0     0  1.5   6.5       0
4    1.20    291     54     6     25    291       0     0  2.1   8.6       0
5    1.21    440     37     9     25    431       0     0  2.0   5.7       0
6    1.22    574    393    17     25    557       0     0  3.0   4.4       0
7    1.23    835   1072    25     34    776       0     0  3.0   4.1       0
8    1.24   1297   1965    41     38   1218       0     0  3.2   2.9       0
9    1.25   1985   2684    56     49   1880     324     0  2.8   2.5       0
10   1.26   2761   5794    80     51   2630     461     0  2.9   1.8       0
11   1.27   4535   6973   106     60   4369     976     0  2.3   1.3       0
12   1.28   5997   9239   132    103   5762    1239     0  2.2   1.7       0
13   1.29   7736  12167   170    124   7442    1370     0  2.2   1.6       0
14   1.30   9720  15238   213    171   9336    1527     0  2.2   1.8       0
15   1.31  11821  17988   259    243  11319    1795     0  2.2   2.1       0
16   2.01  14411  19544   304    328  13779    2110     0  2.1   2.3       0
17   2.02  17238  21558   361    475  16402    2296     0  2.1   2.8       0
18   2.03  20471  23214   425    632  19414    2788     0  2.1   3.1       0
19   2.04  24363  23260   491    892  22980    3219     0  2.0   3.7       0
20   2.05  28060  24702   564   1153  26343    3859     0  2.0   4.1       0
21   2.06  31211  26359   637   1542  29032    4821     0  2.0   4.9       0
22   2.07  34598  27657   723   2052  31823    6101     0  2.1   5.9       0
23   2.08  37251  28942   812   2651  33788    6188     0  2.2   7.1       0
24   2.09  40235  23589   909   3283  36043    6484     0  2.3   8.2       0
25   2.10  42708  21675  1017   3998  37693    7333     0  2.4   9.4       0
26   2.11  44730  16067  1114   4742  38874    8204     0  2.5  10.6       0
27   2.12  59882  13435  1368   5915  52599    8030     0  2.3   9.9       0
28   2.13  63932  10109  1381   6728  55823   10204     0  2.2  10.5       0
29   2.14  66576   8969  1524   8101  56951   11053     0  2.3  12.2       0
30   2.15  68584   8228  1666   9425  57493   11272     0  2.4  13.7       0
31   2.16  70635   7264  1772  10853  58010   10644     0  2.5  15.4       0
32   2.17  72528   6242  1870  12561  58097   11741     0  2.6  17.3       0
33   2.18  74279   5248  2006  14387  57886   11977     0  2.7  19.4       0
34   2.19  75002   4922  2121  16157  56386   11864     0  2.8  21.5       0
35   2.20  75993   5206  2239  18266  55051   11633     0  2.9  24.0       0
36   2.21  76392   5365  2348  20673  53371   11477     0  3.1  27.1       0
37   2.22  77041   4148  2445  22907  51689   10968     0  3.2  29.7       0
38   2.23  77262   3434  2595  24757  49910    9915     0  3.4  32.0       0
39   2.24  77779   2824  2666  27353  47760    9126     0  3.4  35.2       0
40   2.25  78190   2491  2718  29775  45697    8752     0  3.5  38.1       0
41   2.26  78630   2358  2747  32531  43352    8346     0  3.5  41.4       0
42   2.27  78959   2308  2791  36157  40011    7952     0  3.5  45.8       0
43   2.28  79389   1418  2838  39049  37502    7664     0  3.6  49.2       0
44   2.29  79968    851  2873  41675  35420    7365     0  3.6  52.1       0
45   3.01  80174    715  2915  44518  32741    7110     0  3.6  55.5       0
46   3.02  80302    587  2946  47260  30096    6806     0  3.7  58.9       0
47   3.03  80422    520  2984  49914  27524    6416     0  3.7  62.1       0
48   3.04  80565    522  3015  52109  25441    5952    20  3.7  64.7       0
49   3.05  80710    482  3045  53793  23872    5737    36  3.8  66.6       0
50   3.06  80813    502  3073  55477  22263    5489    60  3.8  68.6       0
51 

In [5]:
# 分析疑似病例随日期的变化
china_day_data.loc[:, ['日期', '疑似']]

日期     疑似
0    1.16      0
1    1.17      0
2    1.18      0
3    1.19      0
4    1.20     54
5    1.21     37
6    1.22    393
7    1.23   1072
8    1.24   1965
9    1.25   2684
10   1.26   5794
11   1.27   6973
12   1.28   9239
13   1.29  12167
14   1.30  15238
15   1.31  17988
16   2.01  19544
17   2.02  21558
18   2.03  23214
19   2.04  23260
20   2.05  24702
21   2.06  26359
22   2.07  27657
23   2.08  28942
24   2.09  23589
25   2.10  21675
26   2.11  16067
27   2.12  13435
28   2.13  10109
29   2.14   8969
30   2.15   8228
31   2.16   7264
32   2.17   6242
33   2.18   5248
34   2.19   4922
35   2.20   5206
36   2.21   5365
37   2.22   4148
38   2.23   3434
39   2.24   2824
40   2.25   2491
41   2.26   2358
42   2.27   2308
43   2.28   1418
44   2.29    851
45   3.01    715
46   3.02    587
47   3.03    520
48   3.04    522
49   3.05    482
50   3.06    502
51   3.07    458
52   3.08    421
53   3.09    349
54   3.10    285
55   3.11    253
56   3.12    147
57   3.13    115
58   3.14    113
59   3.15    134
60   3.16    128
61   3.17    119
62   3.18    105
63   3.19    104
64   3.20    106
65   3.21    118
66   3.22    136
67   3.23    132
68   3.24    134
69   3.25    159
70   3.26    189
71   3.27    184
72   3.28    174
73   3.29    168
74   3.30    183
75   3.31    172
76   4.01    153
77   4.02    135
78   4.03    114
79   4.04    107
80   4.05     88
81   4.06     89
82   4.07     83
83   4.08     73
84   4.09     53
85   4.10     44
86   4.11     82
87   4.12     72
88   4.13     72
89   4.14     73
90   4.15     63
91   4.16     62
92   4.17     63
93   4.18     48
94   4.19     43
95   4.20     37
96   4.21     35
97   4.22     20
98   4.23     20
99   4.24     17
100  4.25     12
101  4.26     10
102  4.27      9
103  4.28     10
104  4.29     10
105  4.30      9
106  5.01     11
107  5.02     10
108  5.03      3
109  5.04      2
110  5.05      5
111  5.06      4
112  5.07      6
113  5.08      8
114  5.09      4
115  5.10      3
116  5.11      3
117  5.12      4
118  5.13      4
119  5.14      4
120  5.15      3
121  5.16      4
122  5.17      4
123  5.18      3
124  5.19      7
125  5.20      7
126  5.21      7
127  5.22      6
128  5.23      9
129  5.24      6
130  5.25      5
131  5.26      6
132  5.27      5
133  5.28      5
134  5.29      5
135  5.30      4
136  5.31      3
137  6.01      2
138  6.02      3
139  6.03      3
140  6.04      2
141  6.05      2
142  6.06      3
143  6.07      4
144  6.08      1
145  6.09      2
146  6.10      1
147  6.11      1
148  6.12      1
149  6.13      2
150  6.14      3
151  6.15      4
152  6.16      7
153  6.17      7
154  6.18      7
155  6.19     11
156  6.20     13
157  6.21     15
158  6.22     15
159  6.23     18
160  6.24     13
161  6.25     10
162  6.26      8
163  6.27      8
164  6.28     10
165  6.29      7
166  6.30      8
167  7.01      5
168  7.02      6
169  7.03      7
170  7.04      7
171  7.05      7
172  7.06      7
173  7.07      6
174  7.08      5
175  7.09      8
176  7.10      8
177  7.11      7
178  7.12      7
179  7.13      5
180  7.14      3
181  7.15      3
182  7.16      3
183  7.17      4
184  7.18      4
185  7.19      4
186  7.20      1
187  7.21      1

+ 可以看到从**1月23日**开始疑似病例逐渐增多，这说明**新冠病毒开始扩散传染了**
+ 而且从**1月20日**有疑似病例开始到**1月28日**短短不到10天就疑似病例就接近**万**人
+ 随后到**2月8日**，疑似病例就达到了**峰值**，这段时间我国重视疫情加大检测，越来越多的疑似病例被检测处理
+ 但是随后，疑似病例逐渐减少，这说明新冠病毒的传播得到了控制
+ 到**2月23日**，疑似病例已经回到了**3千多**，而且仍在下降
+ 最后**三月份三位数的疑似**，**四月份两位数的疑似**，**五月份一位数的疑似**，疫情短短不都两三个月就得到了控制
+ 但到**六月份**，疑似病例又开始增多，但也只是**从个位数到十位数**，这是因为**境外输入病例**传染的还有一些是**进口食品**上携带的
+ 不过也得到了控制，到现在疑似病例被控制在了个位数或十位数

In [6]:
# 分析现有确诊随日期的变化
china_day_data.loc[:, ['日期', '现有确诊']]

日期   现有确诊
0    1.16      0
1    1.17      0
2    1.18      0
3    1.19      0
4    1.20    291
5    1.21    431
6    1.22    557
7    1.23    776
8    1.24   1218
9    1.25   1880
10   1.26   2630
11   1.27   4369
12   1.28   5762
13   1.29   7442
14   1.30   9336
15   1.31  11319
16   2.01  13779
17   2.02  16402
18   2.03  19414
19   2.04  22980
20   2.05  26343
21   2.06  29032
22   2.07  31823
23   2.08  33788
24   2.09  36043
25   2.10  37693
26   2.11  38874
27   2.12  52599
28   2.13  55823
29   2.14  56951
30   2.15  57493
31   2.16  58010
32   2.17  58097
33   2.18  57886
34   2.19  56386
35   2.20  55051
36   2.21  53371
37   2.22  51689
38   2.23  49910
39   2.24  47760
40   2.25  45697
41   2.26  43352
42   2.27  40011
43   2.28  37502
44   2.29  35420
45   3.01  32741
46   3.02  30096
47   3.03  27524
48   3.04  25441
49   3.05  23872
50   3.06  22263
51   3.07  20616
52   3.08  19097
53   3.09  17802
54   3.10  16226
55   3.11  14920
56   3.12  13607
57   3.13  12178
58   3.14  10822
59   3.15   9996
60   3.16   9086
61   3.17   8184
62   3.18   7438
63   3.19   6763
64   3.20   6013
65   3.21   5849
66   3.22   5483
67   3.23   5165
68   3.24   4768
69   3.25   4471
70   3.26   4043
71   3.27   3790
72   3.28   3435
73   3.29   3199
74   3.30   3006
75   3.31   2895
76   4.01   2787
77   4.02   2686
78   4.03   2556
79   4.04   2382
80   4.05   2317
81   4.06   2281
82   4.07   2229
83   4.08   2194
84   4.09   2122
85   4.10   2085
86   4.11   2113
87   4.12   2106
88   4.13   2083
89   4.14   2004
90   4.15   1941
91   4.16   1872
92   4.17   1794
93   4.18   1734
94   4.19   1700
95   4.20   1630
96   4.21   1603
97   4.22   1513
98   4.23   1433
99   4.24   1320
100  4.25   1246
101  4.26   1140
102  4.27   1040
103  4.28    997
104  4.29    945
105  4.30    897
106  5.01    840
107  5.02    811
108  5.03    741
109  5.04    635
110  5.05    559
111  5.06    498
112  5.07    448
113  5.08    372
114  5.09    299
115  5.10    274
116  5.11    241
117  5.12    220
118  5.13    199
119  5.14    181
120  5.15    166
121  5.16    157
122  5.17    148
123  5.18    147
124  5.19    145
125  5.20    140
126  5.21    139
127  5.22    137
128  5.23    134
129  5.24    134
130  5.25    131
131  5.26    125
132  5.27    116
133  5.28    111
134  5.29    118
135  5.30    119
136  5.31    131
137  6.01    128
138  6.02    133
139  6.03    127
140  6.04    127
141  6.05    127
142  6.06    130
143  6.07    124
144  6.08    118
145  6.09    113
146  6.10    119
147  6.11    118
148  6.12    123
149  6.13    178
150  6.14    220
151  6.15    256
152  6.16    296
153  6.17    314
154  6.18    346
155  6.19    362
156  6.20    382
157  6.21    402
158  6.22    441
159  6.23    460
160  6.24    473
161  6.25    493
162  6.26    509
163  6.27    516
164  6.28    512
165  6.29    525
166  6.30    516
167  7.01    528
168  7.02    528
169  7.03    522
170  7.04    514
171  7.05    515
172  7.06    529
173  7.07    526
174  7.08    511
175  7.09    529
176  7.10    546
177  7.11    560
178  7.12    575
179  7.13    598
180  7.14    621
181  7.15    601
182  7.16    648
183  7.17    697
184  7.18    749
185  7.19    836
186  7.20    894
187  7.21    921

+ 可以看到与疑似病例的情况差不多，**二月份**是在**当时的现有确诊**是**最多**的时候
+ 而且随着疑似病例到三月的逐渐减少，进入三月份**当时的现有现有确诊**也开始**逐渐减少**
+ 这说明，疫情在二月份在国内达到了顶峰，但是随后就得到了控制
+ **当时的现有重症患者**也符合这个情况，都是二月份时是最多的，后来就逐渐减少了

In [7]:
# 分析境外输入随时间的变化
china_day_data.loc[:, ['日期', '境外输入']]

日期  境外输入
0    1.16     0
1    1.17     0
2    1.18     0
3    1.19     0
4    1.20     0
5    1.21     0
6    1.22     0
7    1.23     0
8    1.24     0
9    1.25     0
10   1.26     0
11   1.27     0
12   1.28     0
13   1.29     0
14   1.30     0
15   1.31     0
16   2.01     0
17   2.02     0
18   2.03     0
19   2.04     0
20   2.05     0
21   2.06     0
22   2.07     0
23   2.08     0
24   2.09     0
25   2.10     0
26   2.11     0
27   2.12     0
28   2.13     0
29   2.14     0
30   2.15     0
31   2.16     0
32   2.17     0
33   2.18     0
34   2.19     0
35   2.20     0
36   2.21     0
37   2.22     0
38   2.23     0
39   2.24     0
40   2.25     0
41   2.26     0
42   2.27     0
43   2.28     0
44   2.29     0
45   3.01     0
46   3.02     0
47   3.03     0
48   3.04    20
49   3.05    36
50   3.06    60
51   3.07    63
52   3.08    67
53   3.09    69
54   3.10    79
55   3.11    85
56   3.12    88
57   3.13    95
58   3.14   111
59   3.15   123
60   3.16   143
61   3.17   155
62   3.18   189
63   3.19   228
64   3.20   269
65   3.21   314
66   3.22   353
67   3.23   427
68   3.24   474
69   3.25   541
70   3.26   595
71   3.27   649
72   3.28   693
73   3.29   723
74   3.30   771
75   3.31   806
76   4.01   841
77   4.02   870
78   4.03   888
79   4.04   913
80   4.05   951
81   4.06   983
82   4.07  1042
83   4.08  1103
84   4.09  1141
85   4.10  1183
86   4.11  1280
87   4.12  1378
88   4.13  1464
89   4.14  1500
90   4.15  1534
91   4.16  1549
92   4.17  1566
93   4.18  1575
94   4.19  1583
95   4.20  1587
96   4.21  1610
97   4.22  1616
98   4.23  1618
99   4.24  1629
100  4.25  1634
101  4.26  1636
102  4.27  1639
103  4.28  1660
104  4.29  1664
105  4.30  1670
106  5.01  1671
107  5.02  1672
108  5.03  1675
109  5.04  1676
110  5.05  1678
111  5.06  1680
112  5.07  1680
113  5.08  1681
114  5.09  1683
115  5.10  1690
116  5.11  1691
117  5.12  1692
118  5.13  1692
119  5.14  1692
120  5.15  1698
121  5.16  1700
122  5.17  1704
123  5.18  1707
124  5.19  1708
125  5.20  1709
126  5.21  1711
127  5.22  1711
128  5.23  1713
129  5.24  1724
130  5.25  1731
131  5.26  1732
132  5.27  1734
133  5.28  1734
134  5.29  1738
135  5.30  1740
136  5.31  1756
137  6.01  1761
138  6.02  1762
139  6.03  1763
140  6.04  1768
141  6.05  1771
142  6.06  1776
143  6.07  1780
144  6.08  1783
145  6.09  1786
146  6.10  1797
147  6.11  1803
148  6.12  1808
149  6.13  1827
150  6.14  1837
151  6.15  1845
152  6.16  1856
153  6.17  1860
154  6.18  1864
155  6.19  1868
156  6.20  1869
157  6.21  1876
158  6.22  1885
159  6.23  1888
160  6.24  1893
161  6.25  1895
162  6.26  1899
163  6.27  1902
164  6.28  1907
165  6.29  1918
166  6.30  1918
167  7.01  1920
168  7.02  1923
169  7.03  1925
170  7.04  1931
171  7.05  1934
172  7.06  1942
173  7.07  1949
174  7.08  1958
175  7.09  1962
176  7.10  1964
177  7.11  1971
178  7.12  1979
179  7.13  1982
180  7.14  1988
181  7.15  1989
182  7.16  1998
183  7.17  2004
184  7.18  2007
185  7.19  2012
186  7.20  2015
187  7.21  2020

+ 可以很清晰的看到，在整个三月份之前国内没有境外输入的病例
+ 但是三月份开始，境外输入病例开始一点点增加，而这时我国已经基本上控制了新冠病毒的传播，疑似病例和重症病例都在逐渐减少
+ 我猜测，有些外国人想来中国避难或者治疗。下面可以结合看外国的历史每日数据，就能发现，三月份开始新冠病毒开始在外国传播

In [24]:
# 分析无症状感染者随时间的变化
china_day_data.loc[:, ['日期', '无症状感染者']]

日期  无症状感染者
0    1.16       0
1    1.17       0
2    1.18       0
3    1.19       0
4    1.20       0
5    1.21       0
6    1.22       0
7    1.23       0
8    1.24       0
9    1.25       0
10   1.26       0
11   1.27       0
12   1.28       0
13   1.29       0
14   1.30       0
15   1.31       0
16   2.01       0
17   2.02       0
18   2.03       0
19   2.04       0
20   2.05       0
21   2.06       0
22   2.07       0
23   2.08       0
24   2.09       0
25   2.10       0
26   2.11       0
27   2.12       0
28   2.13       0
29   2.14       0
30   2.15       0
31   2.16       0
32   2.17       0
33   2.18       0
34   2.19       0
35   2.20       0
36   2.21       0
37   2.22       0
38   2.23       0
39   2.24       0
40   2.25       0
41   2.26       0
42   2.27       0
43   2.28       0
44   2.29       0
45   3.01       0
46   3.02       0
47   3.03       0
48   3.04       0
49   3.05       0
50   3.06       0
51   3.07       0
52   3.08       0
53   3.09       0
54   3.10       0
55   3.11       0
56   3.12       0
57   3.13       0
58   3.14       0
59   3.15       0
60   3.16       0
61   3.17       0
62   3.18       0
63   3.19       0
64   3.20       0
65   3.21       0
66   3.22       0
67   3.23       0
68   3.24       0
69   3.25       0
70   3.26       0
71   3.27       0
72   3.28       0
73   3.29       0
74   3.30       0
75   3.31    1367
76   4.01    1075
77   4.02    1027
78   4.03    1030
79   4.04    1024
80   4.05    1047
81   4.06    1033
82   4.07    1095
83   4.08    1104
84   4.09    1097
85   4.10    1092
86   4.11    1086
87   4.12    1064
88   4.13    1005
89   4.14    1023
90   4.15    1032
91   4.16    1038
92   4.17    1017
93   4.18     999
94   4.19     990
95   4.20     992
96   4.21     991
97   4.22     984
98   4.23     979
99   4.24     983
100  4.25    1000
101  4.26     974
102  4.27     997
103  4.28     993
104  4.29     998
105  4.30     981
106  5.01     989
107  5.02     968
108  5.03     962
109  5.04     947
110  5.05     903
111  5.06     880
112  5.07     854
113  5.08     836
114  5.09     794
115  5.10     780
116  5.11     760
117  5.12     750
118  5.13     712
119  5.14     619
120  5.15     561
121  5.16     515
122  5.17     448
123  5.18     389
124  5.19     368
125  5.20     375
126  5.21     365
127  5.22     370
128  5.23     371
129  5.24     396
130  5.25     403
131  5.26     404
132  5.27     413
133  5.28     409
134  5.29     396
135  5.30     389
136  5.31     397
137  6.01     371
138  6.02     357
139  6.03     326
140  6.04     297
141  6.05     257
142  6.06     236
143  6.07     201
144  6.08     174
145  6.09     155
146  6.10     129
147  6.11     104
148  6.12      98
149  6.13     103
150  6.14     112
151  6.15     110
152  6.16     106
153  6.17     111
154  6.18     110
155  6.19     108
156  6.20     111
157  6.21     114
158  6.22      99
159  6.23     100
160  6.24      97
161  6.25      97
162  6.26     106
163  6.27     106
164  6.28     102
165  6.29      99
166  6.30     100
167  7.01      99
168  7.02      97
169  7.03      98
170  7.04      99
171  7.05     109
172  7.06     114
173  7.07     117
174  7.08     112
175  7.09     113
176  7.10     111
177  7.11     112
178  7.12     110
179  7.13     110
180  7.14     110
181  7.15     104
182  7.16     104
183  7.17     109
184  7.18     147
185  7.19     154
186  7.20     149
187  7.21     164

+ 可以很清楚的看到异常的情况，**3月31日之前**无症状感染者都是0，但是**3月31日**无症状感染者就是**1367**人了
+ 通过查看当时的新闻发现，在3月31日之前我国并没有记录无症状感染者人数，而从3月31日才开始正式记录统计无症状感染者人数
+ 这个无症状感染者的人数从记录开始也是逐渐减少，因为3月份我国已经控制住新冠病毒传播，每日确诊人数也在逐渐减少
+ 但是由于**六七月份境外输入和进口食品又传播了病毒**，导致无症状感染者又多了起来，但是也只是百位数，还是被控制住了
+ 彰显了我国的强大！

## 分析国内历史每日新增数据

可以通过国内每日新增数据数据来再次说明问题

In [9]:
# 查看国内历史每日新增数据
china_dayadd_data = pd.read_csv(filepath_cn+"国内历史每日新增数据.csv") # 这其实是每日累计数据
# 列名中文化和调整列顺序
china_dayadd_data.columns = ['新增确诊', '新增疑似', '新增死亡', '新增治愈', '新增境外输入', '新增无症状感染者', '死亡率', '治愈率', '日期']
# 日期列调到第一列
data_col = china_dayadd_data.pop('日期') # 先弹出该列
china_dayadd_data.insert(0, '日期', data_col) # 再将该列添加到第一列即可
china_dayadd_data

日期   新增确诊  新增疑似  新增死亡  新增治愈  新增境外输入  新增无症状感染者   死亡率      治愈率
0    1.23    259   680     8     6       0         0   3.1      2.3
1    1.24    444  1118    16     3       0         0   3.6      0.7
2    1.25    688  1309    15    11       0         0   2.2      1.6
3    1.26    769  3806    24     2       0         0   3.1      0.3
4    1.27   1771  2077    26     9       0         0   1.5      0.5
5    1.28   1459  3248    26    43       0         0   1.8      2.9
6    1.29   1737  4148    38    21       0         0   2.2      1.2
7    1.30   1982  4812    43    47       0         0   2.2      2.4
8    1.31   2102  5019    46    72       0         0   2.2      3.4
9    2.01   2590  4562    45    85       0         0   1.7      3.3
10   2.02   2829  5173    57   147       0         0   2.0      5.2
11   2.03   3235  5072    64   157       0         0   2.0      4.9
12   2.04   3893  3971    65   262       0         0   1.7      6.7
13   2.05   3697  5328    73   261       0         0   2.0      7.1
14   2.06   3143  4833    73   387       0         0   2.3     12.3
15   2.07   3401  4214    86   510       0         0   2.5     15.0
16   2.08   2656  3916    89   600       0         0   3.4     22.6
17   2.09   3062  4008    97   632       0         0   3.2     20.6
18   2.10   2484  3536   108   716       0         0   4.3     28.8
19   2.11   2022  3342    97   744       0         0   4.8     36.8
20   2.12  15153  2807   254  1173       0         0   1.7      7.7
21   2.13   5093  2450   121  1083       0         0   2.4     21.3
22   2.14   2644  2277   143  1373       0         0   5.4     51.9
23   2.15   2009  1918   142  1324       0         0   7.1     65.9
24   2.16   2051  1563   105  1425       0         0   5.1     69.5
25   2.17   1891  1432    98  1701       0         0   5.2     90.0
26   2.18   1751  1185   136  1826       0         0   7.8    104.3
27   2.19    820  1277   115  1781       0         0  14.0    217.2
28   2.20    892  1614   118  2109       0         0  13.2    236.4
29   2.21    399  1361   109  2394       0         0  27.3    600.0
30   2.22    649   882    97  2235       0         0  14.9    344.4
31   2.23    416   620   150  1850       0         0  36.1    444.7
32   2.24    517   530    71  2596       0         0  13.7    502.1
33   2.25    411   439    52  2422       0         0  12.7    589.3
34   2.26    440   508    29  2756       0         0   6.6    626.4
35   2.27    329   452    44  3626       0         0  13.4   1102.1
36   2.28    430   248    47  2892       0         0  10.9    672.6
37   2.29    579   132    35  2626       0         0   6.0    453.5
38   3.01    206   141    42  2843       0         0  20.4   1380.1
39   3.02    128   129    31  2742       0         0  24.2   2142.2
40   3.03    120   143    38  2654       0         0  31.7   2211.7
41   3.04    143   143    31  2195       2         0  21.7   1535.0
42   3.05    145   102    30  1684      16         0  20.7   1161.4
43   3.06    103    99    28  1684      24         0  27.2   1635.0
44   3.07     46    84    27  1666       3         0  58.7   3621.7
45   3.08     45    60    23  1541       4         0  51.1   3424.4
46   3.09     20    36    17  1298       2         0  85.0   6490.0
47   3.10     31    31    22  1585      10         0  71.0   5112.9
48   3.11     25    33    11  1320       6         0  44.0   5280.0
49   3.12     11    33     7  1329       3         0  63.6  12081.8
50   3.13     18    17    14  1433       7         0  77.8   7961.1
51   3.14     27    39    10  1373      16         0  37.0   5085.2
52   3.15     29    41    14   841      12         0  48.3   2900.0
53   3.16     39    45    13   936      20         0  33.3   2400.0
54   3.17     35    21    11   926      12         0  31.4   2645.7
55   3.18     84    23     8   822      34         0   9.5    978.6
56   3.19     65    31     3   737      39         0   4.6   1133.8
57   3.20    116    36     8   592      41         0   6.9    5

In [10]:
# 分析新增确诊随时间的变化
china_dayadd_data.loc[:, ['日期', '新增确诊']]

日期   新增确诊
0    1.23    259
1    1.24    444
2    1.25    688
3    1.26    769
4    1.27   1771
5    1.28   1459
6    1.29   1737
7    1.30   1982
8    1.31   2102
9    2.01   2590
10   2.02   2829
11   2.03   3235
12   2.04   3893
13   2.05   3697
14   2.06   3143
15   2.07   3401
16   2.08   2656
17   2.09   3062
18   2.10   2484
19   2.11   2022
20   2.12  15153
21   2.13   5093
22   2.14   2644
23   2.15   2009
24   2.16   2051
25   2.17   1891
26   2.18   1751
27   2.19    820
28   2.20    892
29   2.21    399
30   2.22    649
31   2.23    416
32   2.24    517
33   2.25    411
34   2.26    440
35   2.27    329
36   2.28    430
37   2.29    579
38   3.01    206
39   3.02    128
40   3.03    120
41   3.04    143
42   3.05    145
43   3.06    103
44   3.07     46
45   3.08     45
46   3.09     20
47   3.10     31
48   3.11     25
49   3.12     11
50   3.13     18
51   3.14     27
52   3.15     29
53   3.16     39
54   3.17     35
55   3.18     84
56   3.19     65
57   3.20    116
58   3.21     82
59   3.22    102
60   3.23    147
61   3.24     99
62   3.25    114
63   3.26    118
64   3.27    135
65   3.28    128
66   3.29    106
67   3.30     98
68   3.31     86
69   4.01     93
70   4.02     78
71   4.03     73
72   4.04     55
73   4.05     75
74   4.06     66
75   4.07     86
76   4.08     92
77   4.09     56
78   4.10     64
79   4.11    113
80   4.12    115
81   4.13     99
82   4.14     49
83   4.15     52
84   4.16     27
85   4.17     31
86   4.18     21
87   4.19     36
88   4.20     13
89   4.21     37
90   4.22     15
91   4.23      9
92   4.24     13
93   4.25     14
94   4.26      3
95   4.27      6
96   4.28     22
97   4.29      4
98   4.30     12
99   5.01      3
100  5.02      5
101  5.03      7
102  5.04      4
103  5.05      2
104  5.06      3
105  5.07      6
106  5.08      1
107  5.09     14
108  5.10     20
109  5.11      1
110  5.12      7
111  5.13      6
112  5.14      5
113  5.15      9
114  5.16      5
115  5.17     10
116  5.18      6
117  5.19      5
118  5.20      2
119  5.21     13
120  5.22      2
121  5.23      3
122  5.24     11
123  5.25      7
124  5.26      1
125  5.27      3
126  5.28      0
127  5.29     18
128  5.30      5
129  5.31     18
130  6.01      9
131  6.02      7
132  6.03      1
133  6.04     11
134  6.05      6
135  6.06      9
136  6.07      5
137  6.08      4
138  6.09      3
139  6.10     11
140  6.11      7
141  6.12     11
142  6.13     58
143  6.14     49
144  6.15     45
145  6.16     44
146  6.17     36
147  6.18     37
148  6.19     30
149  6.20     27
150  6.21     21
151  6.22     52
152  6.23     28
153  6.24     21
154  6.25     29
155  6.26     24
156  6.27     18
157  6.28     14
158  6.29     23
159  6.30      5
160  7.01     31
161  7.02     15
162  7.03      9
163  7.04     19
164  7.05     14
165  7.06     25
166  7.07     21
167  7.08     33
168  7.09     46
169  7.10     42
170  7.11     35
171  7.12     46
172  7.13     55
173  7.14     54
174  7.15     20
175  7.16     78
176  7.17     82
177  7.18     80
178  7.19    131
179  7.20     84
180  7.21     74

+ 光从数据就可以清晰的看到整个2月份每日新增确诊病例变化的趋势是**先上升后下降**
+ 其中比较突出的点是，2月12日，单日新增确诊数据达到了15153人
+ 可以通过**查看新增疑似**来探究为什么该日的新增确诊有这么多人

In [11]:
# 分析新增疑似随时间的变化
china_dayadd_data.loc[:, ['日期', '新增确诊', '新增疑似']]

日期   新增确诊  新增疑似
0    1.23    259   680
1    1.24    444  1118
2    1.25    688  1309
3    1.26    769  3806
4    1.27   1771  2077
5    1.28   1459  3248
6    1.29   1737  4148
7    1.30   1982  4812
8    1.31   2102  5019
9    2.01   2590  4562
10   2.02   2829  5173
11   2.03   3235  5072
12   2.04   3893  3971
13   2.05   3697  5328
14   2.06   3143  4833
15   2.07   3401  4214
16   2.08   2656  3916
17   2.09   3062  4008
18   2.10   2484  3536
19   2.11   2022  3342
20   2.12  15153  2807
21   2.13   5093  2450
22   2.14   2644  2277
23   2.15   2009  1918
24   2.16   2051  1563
25   2.17   1891  1432
26   2.18   1751  1185
27   2.19    820  1277
28   2.20    892  1614
29   2.21    399  1361
30   2.22    649   882
31   2.23    416   620
32   2.24    517   530
33   2.25    411   439
34   2.26    440   508
35   2.27    329   452
36   2.28    430   248
37   2.29    579   132
38   3.01    206   141
39   3.02    128   129
40   3.03    120   143
41   3.04    143   143
42   3.05    145   102
43   3.06    103    99
44   3.07     46    84
45   3.08     45    60
46   3.09     20    36
47   3.10     31    31
48   3.11     25    33
49   3.12     11    33
50   3.13     18    17
51   3.14     27    39
52   3.15     29    41
53   3.16     39    45
54   3.17     35    21
55   3.18     84    23
56   3.19     65    31
57   3.20    116    36
58   3.21     82    45
59   3.22    102    47
60   3.23    147    35
61   3.24     99    33
62   3.25    114    58
63   3.26    118    49
64   3.27    135    29
65   3.28    128    28
66   3.29    106    17
67   3.30     98    44
68   3.31     86    26
69   4.01     93    20
70   4.02     78    12
71   4.03     73    11
72   4.04     55    11
73   4.05     75    10
74   4.06     66    12
75   4.07     86    12
76   4.08     92    17
77   4.09     56     3
78   4.10     64     8
79   4.11    113    49
80   4.12    115     6
81   4.13     99     3
82   4.14     49    11
83   4.15     52     4
84   4.16     27     3
85   4.17     31     5
86   4.18     21     2
87   4.19     36     2
88   4.20     13     3
89   4.21     37     3
90   4.22     15     0
91   4.23      9     2
92   4.24     13     3
93   4.25     14     0
94   4.26      3     5
95   4.27      6     1
96   4.28     22     2
97   4.29      4     3
98   4.30     12     3
99   5.01      3     2
100  5.02      5     0
101  5.03      7     1
102  5.04      4     0
103  5.05      2     3
104  5.06      3     2
105  5.07      6     3
106  5.08      1     2
107  5.09     14     1
108  5.10     20     0
109  5.11      1     1
110  5.12      7     1
111  5.13      6     0
112  5.14      5     1
113  5.15      9     2
114  5.16      5     2
115  5.17     10     1
116  5.18      6     1
117  5.19      5     3
118  5.20      2     1
119  5.21     13     1
120  5.22      2     2
121  5.23      3     3
122  5.24     11     0
123  5.25      7     0
124  5.26      1     1
125  5.27      3     0
126  5.28      0     0
127  5.29     18     1
128  5.30      5     0
129  5.31     18     0
130  6.01      9     0
131  6.02      7     1
132  6.03      1     0
133  6.04     11     0
134  6.05      6     1
135  6.06      9     2
136  6.07      5     1
137  6.08      4     0
138  6.09      3     0
139  6.10     11     0
140  6.11      7     0
141  6.12     11     0
142  6.13     58     1
143  6.14     49     1
144  6.15     45     3
145  6.16     44     3
146  6.17     36     3
147  6.18     37     2
148  6.19     30     4
149  6.20     27     3
150  6.21     21     2
151  6.22     52     2
152  6.23     28     4
153  6.24     21     0
154  6.25     29     3
155  6.26     24     1
156  6.27     18     1
157  6.28     14     4
158  6.29     23     1
159  6.30      5     2
160  7.01     31     0
161  7.02     15     1
162  7.03      9     2
163  7.04     19     1
164  7.05     14     0
165  7.06     25     2
166  7.07     21     0
167  7.08     33     0
168  7.09     46     3
169  7.10     42     0
170  7.11     35     0
171  7.12     46     0
172  7.13     55     0
173  

+ 从1月29日开始到2月11日每天的新增疑似病例都在3000以上
+ 而新冠病毒的隔离观察期是两周
+ 因此，在这两周内甚至更往前几天，这些疑似病例大部分都是感染者，最终检测阳性，成为新增确诊
+ 不过随后新增疑似病例就开始减少了

In [25]:
# 分析新增治愈和新增死亡随时间的变化
china_dayadd_data.loc[:, ['日期', '新增死亡', '新增治愈']]

日期  新增死亡  新增治愈
0    1.23     8     6
1    1.24    16     3
2    1.25    15    11
3    1.26    24     2
4    1.27    26     9
5    1.28    26    43
6    1.29    38    21
7    1.30    43    47
8    1.31    46    72
9    2.01    45    85
10   2.02    57   147
11   2.03    64   157
12   2.04    65   262
13   2.05    73   261
14   2.06    73   387
15   2.07    86   510
16   2.08    89   600
17   2.09    97   632
18   2.10   108   716
19   2.11    97   744
20   2.12   254  1173
21   2.13   121  1083
22   2.14   143  1373
23   2.15   142  1324
24   2.16   105  1425
25   2.17    98  1701
26   2.18   136  1826
27   2.19   115  1781
28   2.20   118  2109
29   2.21   109  2394
30   2.22    97  2235
31   2.23   150  1850
32   2.24    71  2596
33   2.25    52  2422
34   2.26    29  2756
35   2.27    44  3626
36   2.28    47  2892
37   2.29    35  2626
38   3.01    42  2843
39   3.02    31  2742
40   3.03    38  2654
41   3.04    31  2195
42   3.05    30  1684
43   3.06    28  1684
44   3.07    27  1666
45   3.08    23  1541
46   3.09    17  1298
47   3.10    22  1585
48   3.11    11  1320
49   3.12     7  1329
50   3.13    14  1433
51   3.14    10  1373
52   3.15    14   841
53   3.16    13   936
54   3.17    11   926
55   3.18     8   822
56   3.19     3   737
57   3.20     8   592
58   3.21     6   506
59   3.22     9   459
60   3.23     7   458
61   3.24     4   492
62   3.25     6   405
63   3.26     5   541
64   3.27     3   385
65   3.28     5   478
66   3.29     5   337
67   3.30     3   288
68   3.31     7   190
69   4.01     6   195
70   4.02     4   175
71   4.03     4   199
72   4.04     3   226
73   4.05     1   138
74   4.06     0   102
75   4.07     2   136
76   4.08     2   125
77   4.09     1   127
78   4.10     4    97
79   4.11     0    85
80   4.12     2   125
81   4.13     0   117
82   4.14     1   127
83   4.15     0   115
84   4.16     0    96
85   4.17     0   109
86   4.18     0    81
87   4.19     0    70
88   4.20     0   131
89   4.21     0   156
90   4.22     0   105
91   4.23     0    89
92   4.24     0   126
93   4.25     0    88
94   4.26     0   108
95   4.27     0   106
96   4.28     0    65
97   4.29     0    56
98   4.30     0    60
99   5.01     0    60
100  5.02     0    34
101  5.03     0    77
102  5.04     0   110
103  5.05     0    78
104  5.06     0    64
105  5.07     0    56
106  5.08     0    77
107  5.09     0    87
108  5.10     0    45
109  5.11     0    33
110  5.12     0    28
111  5.13     0    27
112  5.14     0    23
113  5.15     0    24
114  5.16     0    14
115  5.17     0    19
116  5.18     0     7
117  5.19     0     7
118  5.20     0     7
119  5.21     0    14
120  5.22     0     4
121  5.23     0     6
122  5.24     0    11
123  5.25     0    10
124  5.26     0     7
125  5.27     0    12
126  5.28     0     5
127  5.29     0    11
128  5.30     0     4
129  5.31     0     6
130  6.01     0    12
131  6.02     0     2
132  6.03     0     7
133  6.04     0    11
134  6.05     0     6
135  6.06     0     6
136  6.07     0    11
137  6.08     0    10
138  6.09     0     8
139  6.10     0     5
140  6.11     0     8
141  6.12     0     7
142  6.13     0     3
143  6.14     0     7
144  6.15     0     9
145  6.16     0     4
146  6.17     0    18
147  6.18     0     5
148  6.19     0    14
149  6.20     1     6
150  6.21     0     1
151  6.22     0    13
152  6.23     1     8
153  6.24     0     8
154  6.25     1     8
155  6.26     0     8
156  6.27     0    11
157  6.28     0    18
158  6.29     0    10
159  6.30     0    14
160  7.01     0    19
161  7.02     0    15
162  7.03     0    15
163  7.04     0    27
164  7.05     0    13
165  7.06     0    11
166  7.07     0    24
167  7.08     0    48
168  7.09     0    28
169  7.10     0    25
170  7.11     0    21
171  7.12     0    31
172  7.13     1    31
173  7.14     0    31
174  7.15     2    38
175  7.16     0    31
176  7.17     1    32
177  7.18     1    27
178  7.19     0    44
179  7.20     0    26
180  7.21     2    45

+ 可以看到在2月12日那几天，新增死亡都在一百左右
+ 但是过了那几天，新增死亡人数逐渐下降，新增治愈人数逐渐上升

In [13]:
# 查看分析历史每日新增数据（湖北、全国和非湖北）
china_dayadd_data2 = pd.read_csv(filepath_cn+"历史每日新增数据（湖北、全国、和非湖北）.csv")
china_dayadd_data2

date  hubei  country  notHubei
0    1.20     72       77         5
1    1.21    105      149        44
2    1.22     69      131        62
3    1.23    105      259       154
4    1.24    180      444       264
5    1.25    323      688       365
6    1.26    371      769       398
7    1.27   1291     1771       480
8    1.28    840     1459       619
9    1.29   1032     1737       705
10   1.30   1220     1982       762
11   1.31   1347     2102       755
12   2.01   1921     2590       669
13   2.02   2103     2829       726
14   2.03   2345     3235       890
15   2.04   3156     3893       737
16   2.05   2987     3697       710
17   2.06   2447     3143       696
18   2.07   2841     3401       560
19   2.08   2147     2656       509
20   2.09   2618     3062       444
21   2.10   2097     2484       387
22   2.11   1638     2022       384
23   2.12  14840    15153       313
24   2.13   4823     5093       270
25   2.14   2420     2644       224
26   2.15   1843     2009       166
27   2.16   1933     2051       118
28   2.17   1807     1891        84
29   2.18   1693     1751        58
30   2.19    775      820        45
31   2.20    631      892       261
32   2.21    366      399        33
33   2.22    630      649        19
34   2.23    398      416        18
35   2.24    499      517        18
36   2.25    401      411        10
37   2.26    409      440        31
38   2.27    318      329        11
39   2.28    423      430         7
40   2.29    570      579         9
41   3.01    196      206        10
42   3.02    114      128        14
43   3.03    115      120         5
44   3.04    134      143         9
45   3.05    126      145        19
46   3.06     74      103        29
47   3.07     41       46         5
48   3.08     36       45         9
49   3.09     17       20         3
50   3.10     13       31        18
51   3.11      8       25        17
52   3.12      5       11         6
53   3.13      4       18        14
54   3.14      4       27        23
55   3.15      4       29        25
56   3.16      4       39        35
57   3.17      1       35        34
58   3.18      0       84        84
59   3.19      0       65        65
60   3.20      0      116       116
61   3.21      0       82        82
62   3.22      0      102       102
63   3.23      1      147       146
64   3.24      0       99        99
65   3.25      0      114       114
66   3.26      0      118       118
67   3.27      0      135       135
68   3.28      0      128       128
69   3.29      0      106       106
70   3.30      0       98        98
71   3.31      1       86        85
72   4.01      0       93        93
73   4.02      0       78        78
74   4.03      1       73        72
75   4.04      0       55        55
76   4.05      0       75        75
77   4.06      0       66        66
78   4.07      0       86        86
79   4.08      0       92        92
80   4.09      0       56        56
81   4.10      0       64        64
82   4.11      0      113       113
83   4.12      0      115       115
84   4.13      0       99        99
85   4.14      0       49        49
86   4.15      0       52        52
87   4.16      0       27        27
88   4.17      0       31        31
89   4.18      0       21        21
90   4.19      0       36        36
91   4.20      0       13        13
92   4.21      0       37        37
93   4.22      0       15        15
94   4.23      0        9         9
95   4.24      0       13        13
96   4.25      0       14        14
97   4.26      0        3         3
98   4.27      0        6         6
99   4.28      0       22        22
100  4.29      0        4         4
101  4.30      0       12        12
102  5.01      0        3         3
103  5.02      0        5         5
104  5.03      0        7         7
105  5.04      0        4         4
106  5.05      0        2         2
107  5.06      0        3         3
108  5.07      0        6         6
109  5.08      0        1         1
110  5.09

分析很容易看出，湖北是此次疫情的爆发地，趋势跟上边分析的趋势差不多，但是在进入六七月以后，非湖北的每日新增人数要多于湖北。
通过新闻得知湖北已经开放了，不咋如疫情期间似的管理那么严，这是胜利的曙光。

# 外国数据分析及处理

## 分析全球历史每日数据

In [14]:
foreign_day_data = pd.read_csv(filepath_fg+"全球历史每日数据.csv") # 都是累计的数据
# 列名中文化
foreign_day_data.columns = ['日期', '累计确诊', '累计死亡', '累计治愈', '新增确诊', '死亡率', '治愈率']
foreign_day_data

日期      累计确诊    累计死亡     累计治愈    新增确诊   死亡率    治愈率
0    1.28        57       0        3       0  0.00   5.26
1    1.29        74       0        3      13  0.00   4.05
2    1.30        98       0        6      22  0.00   6.12
3    1.31       124       0       11      20  0.00   8.87
4    2.01       139       0       11      13  0.00   7.91
5    2.02       149       1       11      10  0.67   7.38
6    2.03       154       1       12       5  0.65   7.79
7    2.04       177       1       19      22  0.56  10.73
8    2.05       201       1       21      14  0.50  10.45
9    2.06       225       1       25      24  0.44  11.11
10   2.07       273       1       27      48  0.37   9.89
11   2.08       299       1       29      26  0.33   9.70
12   2.09       313       1       30      14  0.32   9.58
13   2.10       385       1       36      72  0.26   9.35
14   2.11       397       1       44      12  0.25  11.08
15   2.12       444       1       50      47  0.23  11.26
16   2.13       505       2       56      61  0.40  11.09
17   2.14       516       2       59      11  0.39  11.43
18   2.15       601       3       61      85  0.50  10.15
19   2.16       687       3       84      78  0.44  12.23
20   2.17       800       3       88     104  0.38  11.00
21   2.18       904       3      114     104  0.33  12.61
22   2.19      1018       3      127     104  0.29  12.48
23   2.20      1096       8      127      78  0.73  11.59
24   2.21      1231      11      142     133  0.89  11.54
25   2.22      1528      14      154     156  0.92  10.08
26   2.23      1895      22      155     265  1.16   8.18
27   2.24      2286      32      173     368  1.40   7.57
28   2.25      2575      40      192     224  1.55   7.46
29   2.26      3085      51      220     492  1.65   7.13
30   2.27      3945      64      310     863  1.62   7.86
31   2.28      4947      80      360     998  1.62   7.28
32   2.29      6305      95      455    1349  1.51   7.22
33   3.01      7342     112     1020    1033  1.53  13.89
34   3.02      9628     152      558    2285  1.58   5.80
35   3.03     12047     184      895    2461  1.53   7.43
36   3.04     13897     236      909    2226  1.70   6.54
37   3.05     16118     294     1087    2534  1.82   6.74
38   3.06     19378     365     1556    3691  1.88   8.03
39   3.07     22903     451     1720    3443  1.97   7.51
40   3.08     26959     561     3208    3403  2.08  11.90
41   3.09     30814     766     3614    4913  2.49  11.73
42   3.10     35796     977     4212    5018  2.73  11.77
43   3.11     41212    1232     5274    5598  2.99  12.80
44   3.12     48743    1577     5779    7810  3.24  11.86
45   3.13     58493    1944     6544    9780  3.32  11.19
46   3.14     67414    2354     7947    9359  3.49  11.79
47   3.15     79559    2765     9304   13683  3.48  11.69
48   3.16     92093    3452    10413   16374  3.75  11.31
49   3.17    106167    4276    11680   14567  4.03  11.00
50   3.18    122946    4988    13496   17844  4.06  10.98
51   3.19    147281    6078    15402   24409  4.13  10.46
52   3.20    169551    6952    17687   23342  4.10  10.43
53   3.21    204426    8697    21255   35240  4.25  10.40
54   3.22    235356   10399    23726   34269  4.42  10.08
55   3.23    270475   12134    26726   35265  4.49   9.88
56   3.24    313748   14007    29315   45222  4.46   9.34
57   3.25    356275   16408    35350   44440  4.61   9.92
58   3.26    407822   18883    41244   51921  4.63  10.11
59   3.27    466847   21854    50313   58846  4.68  10.78
60   3.28    537400   25313    59270   70494  4.71  11.03
61   3.29    600269   28748    67976   63740  4.79  11.32
62   3.30    659449   31891    78354   60337  4.84  11.88
63   3.31    723899   36231    96623   64446  5.00  13.35
64   4.01    802883   40894   109027   78982  5.09  13.58
65   4.02    882341   45906   126381   79458  5.20  14.32
66   4.03    958134   51798   144442   75792  5.41  15.08
67   4.04   1057294   57549   156928   99160  5.44  14.84
68   

+ 单独看日期和确诊，可以发现，在我国2月份的疫情爆发期，外国整体的数据都还趋于平稳
+ 但是进入三月份后外国的确诊病例就在突飞猛进
+ 短短三周不到，确诊病例就突破10万，接着不到一个星期就破20万
+ 随后确诊人数越来越多，到4月4日突破100万，5月21日突破500万，6月28日突破1000万
+ 到代码敲好的7月21日，已经外国累计1482多万人确诊

In [15]:
# 国外新增确诊和国内新增确诊随时间变化的对比
fg_tmp = foreign_day_data.loc[:, ['日期', '新增确诊']]
fg_tmp['新增确诊（国内）'] = china_dayadd_data['新增确诊']
fg_tmp

日期    新增确诊  新增确诊（国内）
0    1.28       0       259
1    1.29      13       444
2    1.30      22       688
3    1.31      20       769
4    2.01      13      1771
5    2.02      10      1459
6    2.03       5      1737
7    2.04      22      1982
8    2.05      14      2102
9    2.06      24      2590
10   2.07      48      2829
11   2.08      26      3235
12   2.09      14      3893
13   2.10      72      3697
14   2.11      12      3143
15   2.12      47      3401
16   2.13      61      2656
17   2.14      11      3062
18   2.15      85      2484
19   2.16      78      2022
20   2.17     104     15153
21   2.18     104      5093
22   2.19     104      2644
23   2.20      78      2009
24   2.21     133      2051
25   2.22     156      1891
26   2.23     265      1751
27   2.24     368       820
28   2.25     224       892
29   2.26     492       399
30   2.27     863       649
31   2.28     998       416
32   2.29    1349       517
33   3.01    1033       411
34   3.02    2285       440
35   3.03    2461       329
36   3.04    2226       430
37   3.05    2534       579
38   3.06    3691       206
39   3.07    3443       128
40   3.08    3403       120
41   3.09    4913       143
42   3.10    5018       145
43   3.11    5598       103
44   3.12    7810        46
45   3.13    9780        45
46   3.14    9359        20
47   3.15   13683        31
48   3.16   16374        25
49   3.17   14567        11
50   3.18   17844        18
51   3.19   24409        27
52   3.20   23342        29
53   3.21   35240        39
54   3.22   34269        35
55   3.23   35265        84
56   3.24   45222        65
57   3.25   44440       116
58   3.26   51921        82
59   3.27   58846       102
60   3.28   70494       147
61   3.29   63740        99
62   3.30   60337       114
63   3.31   64446       118
64   4.01   78982       135
65   4.02   79458       128
66   4.03   75792       106
67   4.04   99160        98
68   4.05   86231        86
69   4.06   62658        93
70   4.07   75564        78
71   4.08   85205        73
72   4.09   83351        55
73   4.10   88336        75
74   4.11   99043        66
75   4.12   79280        86
76   4.13   71198        92
77   4.14   70194        56
78   4.15   75936        64
79   4.16   82740       113
80   4.17   85816       115
81   4.18   85709        99
82   4.19   79164        49
83   4.20   70814        52
84   4.21   74333        27
85   4.22   86197        31
86   4.23   78989        21
87   4.24   96599        36
88   4.25   96968        13
89   4.26   71399        37
90   4.27   69546        15
91   4.28   73895         9
92   4.29   82600        13
93   4.30   83860        14
94   5.01   86783         3
95   5.02   90049         6
96   5.03   57911        22
97   5.04  106412         4
98   5.05   80972        12
99   5.06   82916         3
100  5.07   98929         5
101  5.08   97216         7
102  5.09   94660         4
103  5.10   80552         2
104  5.11   80462         3
105  5.12   81225         6
106  5.13   83547         1
107  5.14   94906        14
108  5.15   98069        20
109  5.16   92230         1
110  5.17   86331         7
111  5.18   86509         6
112  5.19   86980         5
113  5.20   94569         9
114  5.21  108771         5
115  5.22   96101        10
116  5.23  118665         6
117  5.24   90202         5
118  5.25  102853         2
119  5.26   82834        13
120  5.27   97778         2
121  5.28  108439         3
122  5.29  123030        11
123  5.30  116271         7
124  5.31  133106         1
125  6.01  103448         3
126  6.02  104999         0
127  6.03   69599        18
128  6.04  171930         5
129  6.05  135837        18
130  6.06  125094         9
131  6.07  129322         7
132  6.08  106793         1
133  6.09  105928        11
134  6.10  103825         6
135  6.11  155732         9
136  6.12  144586         5
137  6.13  138084         4
138  6.14  126736         3
139  6.15  126973        11
140  6.16  128161         7
141  6.17  138346        11
142

+ 同样在二月份国内的新增确诊是最多的，但是国外是平缓的状态
+ 而进入三月份后，外国新增确诊开始越来越多，而国内开始逐渐减少
+ 但是当我国控制住疫情传播后，却没有完全控制住境外输入
+ 因此在这几个月仍然每天都有十几例几十例新增确诊
+ 然而外国每日新增确诊仍没有下降的趋势，仍处于一个高水平的状态
+ 可推测外国疫情仍会继续蔓延下去，短时间内不会好转

## 分析外国疫情数据（按国家或大洲查看）

In [16]:
foreign_data = pd.read_csv(filepath_fg+"外国疫情数据（按国家或大洲查看）.csv", encoding='GBK') # 都是累计的数据
# 去除不要的列，并且列名中文化
foreign_data = foreign_data.iloc[:, :11] # 后面的较前一日比较的数据就不用看了
del foreign_data['isUpdated']
del foreign_data['suspect']
del foreign_data['confirmAddCut']
foreign_data.columns=['国家', '大洲', '日期', '新增确诊', '累计确诊', '累计死亡', '累计治愈', '现有确诊']
foreign_data

国家   大洲    日期   新增确诊     累计确诊    累计死亡     累计治愈     现有确诊
0            美国  北美洲  7.22  67140  4028569  144953  1886583  1997033
1            巴西  南美洲  7.22  41008  2159654   81487  1563382   514785
2            印度   亚洲  7.22  37724  1193078   28732   753050   411296
3           俄罗斯   欧洲  7.22   5806   787846   12722   570984   204140
4            南非   非洲  7.22   8170   381798    5368   208144   168286
5            秘鲁  南美洲  7.22   8497   362087   13579   248746    99762
6           墨西哥  北美洲  7.22   6859   356255   40400   267407    48448
7            智利  南美洲  7.22   3753   334683    8677   306816    19190
8           西班牙   欧洲  7.22   1358   313274   28424   196958    87892
9            英国   欧洲  7.22      0   297389   45507     1414   250468
10           伊朗   亚洲  7.22   2586   281413   14853   244840    21720
11         巴基斯坦   亚洲  7.22   1332   267428    5677   210468    51283
12        沙特阿拉伯   亚洲  7.22      0   255825    2557   207259    46009
13          意大利   欧洲  7.22      0   244752   35073   197431    12248
14          土耳其   亚洲  7.22    928   221500    5526   204011    11963
15           法国   欧洲  7.22    584   214607   30168    79861   104578
16          孟加拉   亚洲  7.22   2744   213254    2751   117202    93301
17         哥伦比亚  南美洲  7.22   7033   211038    7166    98840   105032
18           德国   欧洲  7.22    438   204003    9102   188575     6326
19          阿根廷  南美洲  7.22   5344   136118    2490    58598    75030
20          加拿大  北美洲  7.22    535   113473    8908    99248     5317
21          卡塔尔   亚洲  7.22    441   107871     163   104641     3067
22          伊拉克   亚洲  7.22      0    97159    3950    64950    28259
23        印度尼西亚   亚洲  7.22   1882    91751    4459    50255    37037
24           埃及   非洲  7.22    676    89078    4399    29473    55206
25           瑞典   欧洲  7.22    338    78504    5667        0    72837
26         厄瓜多尔  南美洲  7.22   1597    76217    5366    32725    38126
27        哈萨克斯坦   亚洲  7.22   1685    75153     585    46790    27778
28          菲律宾   亚洲  7.22   1505    72269    1843    23623    46803
29           阿曼   亚洲  7.22   1660    71547     349    47922    23276
30         白俄罗斯   欧洲  7.22    173    66521     513    59061     6947
31          比利时   欧洲  7.22    164    64258    9805    17330    37123
32         玻利维亚  南美洲  7.22   1366    62357    2273    19290    40794
33          乌克兰   欧洲  7.22    841    62295    1553    34176    26566
34          科威特   亚洲  7.22      0    60434     412    50919     9103
35          阿联酋   亚洲  7.22      0    57498     341    49964     7193
36          巴拿马  北美洲  7.22    727    55153    1159    30075    23919
37         多米尼加  北美洲  7.22      0    54797     999    25976    27822
38          以色列   亚洲  7.22   1976    54663     430    22920    31313
39           荷兰   欧洲  7.22    170    52475    6158      194    46123
40          葡萄牙   欧洲  7.22      0    48898    1697    33769    13432
41          新加坡   亚洲  7.22    310    48744      27    44584     4133
42           波兰   欧洲  7.22    380    41162    1642    31139     8381
43         危地马拉  北美洲  7.22   1190    40229    1531    26685    12013
44         罗马尼亚   欧洲  7.22   1030    40163    2101    24663    13399
45         尼日利亚   非洲  7.22    576    37801     805    15677    21319
46           巴林   亚洲  7.22    380    37316     130    33455     3731
47          阿富汗   亚洲  7.22    112    35727    1190    23924    10613
48         亚美尼亚   亚洲  7.22    439    35693     678    24766    10249
49         洪都拉斯  北美洲  7.22    734    35345     988     4144    30213
50           瑞士   欧洲  7.22    141    33883    1972    30500     1411
51           加纳   非洲  7.22      0    28989     153    25331     3505
52       吉尔吉斯斯坦   亚洲  7.22    729    28980    1111    15536    12333
53         阿塞拜疆   亚洲  7.22    352    28242     376    19939     7927
54         日本本土   亚洲  7.22    698    27026     988    20001     6037
55          爱尔兰   欧洲  7.22     36    25802    1753    23364      685
56        阿尔及利亚   非洲  7.22    587    24278    1100    16646     6532
57         

In [17]:
foreign_data.sort_values(by=['累计确诊'], ascending = False) # 按累计确诊降序排列分析数据

国家   大洲    日期   新增确诊     累计确诊    累计死亡     累计治愈     现有确诊
0            美国  北美洲  7.22  67140  4028569  144953  1886583  1997033
1            巴西  南美洲  7.22  41008  2159654   81487  1563382   514785
2            印度   亚洲  7.22  37724  1193078   28732   753050   411296
3           俄罗斯   欧洲  7.22   5806   787846   12722   570984   204140
4            南非   非洲  7.22   8170   381798    5368   208144   168286
5            秘鲁  南美洲  7.22   8497   362087   13579   248746    99762
6           墨西哥  北美洲  7.22   6859   356255   40400   267407    48448
7            智利  南美洲  7.22   3753   334683    8677   306816    19190
8           西班牙   欧洲  7.22   1358   313274   28424   196958    87892
9            英国   欧洲  7.22      0   297389   45507     1414   250468
10           伊朗   亚洲  7.22   2586   281413   14853   244840    21720
11         巴基斯坦   亚洲  7.22   1332   267428    5677   210468    51283
12        沙特阿拉伯   亚洲  7.22      0   255825    2557   207259    46009
13          意大利   欧洲  7.22      0   244752   35073   197431    12248
14          土耳其   亚洲  7.22    928   221500    5526   204011    11963
15           法国   欧洲  7.22    584   214607   30168    79861   104578
16          孟加拉   亚洲  7.22   2744   213254    2751   117202    93301
17         哥伦比亚  南美洲  7.22   7033   211038    7166    98840   105032
18           德国   欧洲  7.22    438   204003    9102   188575     6326
19          阿根廷  南美洲  7.22   5344   136118    2490    58598    75030
20          加拿大  北美洲  7.22    535   113473    8908    99248     5317
21          卡塔尔   亚洲  7.22    441   107871     163   104641     3067
22          伊拉克   亚洲  7.22      0    97159    3950    64950    28259
23        印度尼西亚   亚洲  7.22   1882    91751    4459    50255    37037
24           埃及   非洲  7.22    676    89078    4399    29473    55206
25           瑞典   欧洲  7.22    338    78504    5667        0    72837
26         厄瓜多尔  南美洲  7.22   1597    76217    5366    32725    38126
27        哈萨克斯坦   亚洲  7.22   1685    75153     585    46790    27778
28          菲律宾   亚洲  7.22   1505    72269    1843    23623    46803
29           阿曼   亚洲  7.22   1660    71547     349    47922    23276
30         白俄罗斯   欧洲  7.22    173    66521     513    59061     6947
31          比利时   欧洲  7.22    164    64258    9805    17330    37123
32         玻利维亚  南美洲  7.22   1366    62357    2273    19290    40794
33          乌克兰   欧洲  7.22    841    62295    1553    34176    26566
34          科威特   亚洲  7.22      0    60434     412    50919     9103
35          阿联酋   亚洲  7.22      0    57498     341    49964     7193
36          巴拿马  北美洲  7.22    727    55153    1159    30075    23919
37         多米尼加  北美洲  7.22      0    54797     999    25976    27822
38          以色列   亚洲  7.22   1976    54663     430    22920    31313
39           荷兰   欧洲  7.22    170    52475    6158      194    46123
40          葡萄牙   欧洲  7.22      0    48898    1697    33769    13432
41          新加坡   亚洲  7.22    310    48744      27    44584     4133
42           波兰   欧洲  7.22    380    41162    1642    31139     8381
43         危地马拉  北美洲  7.22   1190    40229    1531    26685    12013
44         罗马尼亚   欧洲  7.22   1030    40163    2101    24663    13399
45         尼日利亚   非洲  7.22    576    37801     805    15677    21319
46           巴林   亚洲  7.22    380    37316     130    33455     3731
47          阿富汗   亚洲  7.22    112    35727    1190    23924    10613
48         亚美尼亚   亚洲  7.22    439    35693     678    24766    10249
49         洪都拉斯  北美洲  7.22    734    35345     988     4144    30213
50           瑞士   欧洲  7.22    141    33883    1972    30500     1411
51           加纳   非洲  7.22      0    28989     153    25331     3505
52       吉尔吉斯斯坦   亚洲  7.22    729    28980    1111    15536    12333
53         阿塞拜疆   亚洲  7.22    352    28242     376    19939     7927
54         日本本土   亚洲  7.22    698    27026     988    20001     6037
55          爱尔兰   欧洲  7.22     36    25802    1753    23364      685
56        阿尔及利亚   非洲  7.22    587    24278    1100    16646     6532
57         

+ 很明显的看出，排名第一二的美国和巴西累计确诊远超其他国家每日新增的病例也是很多
+ 这里有个很明显的点，就是印度和俄罗斯还有一些其他，新增确诊病例为0，但这不符合实际情况
+ 应该是在爬取数据的时间点，这些国家的统计数据还没发布出来
+ 所以就存在有的国家明显还是会有新增确诊病例，但是却为0

In [18]:
foreign_data.sort_values(by=['现有确诊'], ascending = False) # 按现有确诊降序排列分析数据

国家   大洲    日期   新增确诊     累计确诊    累计死亡     累计治愈     现有确诊
0            美国  北美洲  7.22  67140  4028569  144953  1886583  1997033
1            巴西  南美洲  7.22  41008  2159654   81487  1563382   514785
2            印度   亚洲  7.22  37724  1193078   28732   753050   411296
9            英国   欧洲  7.22      0   297389   45507     1414   250468
3           俄罗斯   欧洲  7.22   5806   787846   12722   570984   204140
4            南非   非洲  7.22   8170   381798    5368   208144   168286
17         哥伦比亚  南美洲  7.22   7033   211038    7166    98840   105032
15           法国   欧洲  7.22    584   214607   30168    79861   104578
5            秘鲁  南美洲  7.22   8497   362087   13579   248746    99762
16          孟加拉   亚洲  7.22   2744   213254    2751   117202    93301
8           西班牙   欧洲  7.22   1358   313274   28424   196958    87892
19          阿根廷  南美洲  7.22   5344   136118    2490    58598    75030
25           瑞典   欧洲  7.22    338    78504    5667        0    72837
24           埃及   非洲  7.22    676    89078    4399    29473    55206
11         巴基斯坦   亚洲  7.22   1332   267428    5677   210468    51283
6           墨西哥  北美洲  7.22   6859   356255   40400   267407    48448
28          菲律宾   亚洲  7.22   1505    72269    1843    23623    46803
39           荷兰   欧洲  7.22    170    52475    6158      194    46123
12        沙特阿拉伯   亚洲  7.22      0   255825    2557   207259    46009
32         玻利维亚  南美洲  7.22   1366    62357    2273    19290    40794
26         厄瓜多尔  南美洲  7.22   1597    76217    5366    32725    38126
31          比利时   欧洲  7.22    164    64258    9805    17330    37123
23        印度尼西亚   亚洲  7.22   1882    91751    4459    50255    37037
38          以色列   亚洲  7.22   1976    54663     430    22920    31313
49         洪都拉斯  北美洲  7.22    734    35345     988     4144    30213
22          伊拉克   亚洲  7.22      0    97159    3950    64950    28259
37         多米尼加  北美洲  7.22      0    54797     999    25976    27822
27        哈萨克斯坦   亚洲  7.22   1685    75153     585    46790    27778
33          乌克兰   欧洲  7.22    841    62295    1553    34176    26566
36          巴拿马  北美洲  7.22    727    55153    1159    30075    23919
29           阿曼   亚洲  7.22   1660    71547     349    47922    23276
10           伊朗   亚洲  7.22   2586   281413   14853   244840    21720
45         尼日利亚   非洲  7.22    576    37801     805    15677    21319
57         塞尔维亚   欧洲  7.22      0    21605     491        0    21114
7            智利  南美洲  7.22   3753   334683    8677   306816    19190
40          葡萄牙   欧洲  7.22      0    48898    1697    33769    13432
44         罗马尼亚   欧洲  7.22   1030    40163    2101    24663    13399
52       吉尔吉斯斯坦   亚洲  7.22    729    28980    1111    15536    12333
13          意大利   欧洲  7.22      0   244752   35073   197431    12248
43         危地马拉  北美洲  7.22   1190    40229    1531    26685    12013
14          土耳其   亚洲  7.22    928   221500    5526   204011    11963
47          阿富汗   亚洲  7.22    112    35727    1190    23924    10613
48         亚美尼亚   亚洲  7.22    439    35693     678    24766    10249
34          科威特   亚洲  7.22      0    60434     412    50919     9103
60       乌兹别克斯坦   亚洲  7.22    569    18324      97     9554     8673
72        哥斯达黎加  北美洲  7.22    277    11811      68     3194     8549
42           波兰   欧洲  7.22    380    41162    1642    31139     8381
53         阿塞拜疆   亚洲  7.22    352    28242     376    19939     7927
66          肯尼亚   非洲  7.22    397    14168     250     6258     7660
76         巴勒斯坦   亚洲  7.22    170     9398      64     1932     7402
35          阿联酋   亚洲  7.22      0    57498     341    49964     7193
30         白俄罗斯   欧洲  7.22    173    66521     513    59061     6947
56        阿尔及利亚   非洲  7.22    587    24278    1100    16646     6532
18           德国   欧洲  7.22    438   204003    9102   188575     6326
54         日本本土   亚洲  7.22    698    27026     988    20001     6037
58         摩尔多瓦   欧洲  7.22      0    21442     708    14856     5878
71         委内瑞拉  南美洲  7.22    440    12774     120     6983     5671
64         

可以发现欧洲和美洲的现有确诊病例很多，尤其是美国、巴西、英国、法国、俄罗斯这些欧洲和美洲的主要国家

In [19]:
# 统计各大洲的数据  先分组再合并
foreign_data.pop('日期') # 日期不重要，就去掉先
foreign_data_groupby = foreign_data.groupby(by='大洲')
foreign_data_agg = foreign_data_groupby.agg(np.sum)
foreign_data_agg

新增确诊     累计确诊    累计死亡     累计治愈     现有确诊
大洲                                           
亚洲   58485  3456983   78475  2446230   932278
其他       0      712      13      651       48
北美洲  78215  4723038  199752  2360414  2162872
南美洲  69193  3361311  121266  2339484   900561
大洋洲    476    14505     151    10185     4169
欧洲   12785  2749770  199956  1599299   950515
非洲   13702   748518   15693   408604   324221

+ 从累计确诊病例来看，其实亚洲、南美洲和欧洲三者比较接近
+ 但是北美洲就较多了100多万确诊，这确诊人数更多的都是在美国
+ 其他地区是指海上的轮船和军舰

In [20]:
# 分析外国人数与确诊人数的关系
million_confirm = pd.read_csv(filepath_fg+"外国人口与确诊数.csv", encoding='GBK')
million_confirm

nation  confirm  population   rate
0       卡塔尔   104533     2781677  37579
1        巴林    34078     1569439  21713
2        智利   319493    18729160  17059
3       科威特    56174     4137309  13577
4        阿曼    59568     4829483  12334
5       巴拿马    48096     4176873  11515
6      亚美尼亚    32490     2951776  11007
7        美国  3545077   327167434  10836
8        秘鲁   333867    31989256  10437
9        巴西  1926824   209469333   9199
10      卢森堡     5056      607728   8320
11      新加坡    46630     5638676   8270
12       瑞典    76001    10183175   7463
13    沙特阿拉伯   237803    33699947   7056
14     白俄罗斯    65269     9485386   6881
15      西班牙   303699    46723749   6500
16      阿联酋    55573     9300000   5976
17     摩尔多瓦    19708     3545883   5558
18      比利时    62781    11422068   5496
19     马尔代夫     2801      515696   5431
20       冰岛     1905      353574   5388
21      爱尔兰    25670     4853506   5289
22      吉布提     4979      958920   5192
23       南非   298292    57779622   5163
24      俄罗斯   738787   144478050   5113
25      以色列    42360     8883800   4768
26      葡萄牙    47051    10281762   4576
27     玻利维亚    50867    11353142   4480
28       英国   292931    66488991   4406
29     多米尼加    46305    10627165   4357
30     厄瓜多尔    69570    17084357   4072
31      意大利   243344    60431283   4027
32       瑞士    33016     8516543   3877
33    哈萨克斯坦    61755    18276499   3379
34       伊朗   262173    81800269   3205
35       法国   209640    66987244   3130
36     哥伦比亚   154277    49648685   3107
37     洪都拉斯    28579     9587522   2981
38       荷兰    51362    17231017   2981
39      加拿大   110350    37058856   2978
40       加蓬     6026     2119275   2843
41     塞尔维亚    18983     6982084   2719
42      土耳其   214993    82319724   2612
43     阿塞拜疆    25113     9942334   2526
44      墨西哥   311486   126190788   2468
45       德国   200456    82927922   2417
46      阿根廷   106910    44494502   2403
47    赤道几内亚     3071     1308974   2346
48       丹麦    13262     5797446   2288
49      奥地利    19021     8847037   2150
50      伊拉克    81757    38433600   2127
51       黑山     1287      622345   2068
52       波黑     6981     3516816   1985
53   吉尔吉斯斯坦    11444     6315800   1812
54     危地马拉    30872    17247807   1790
55     罗马尼亚    33585    19473936   1725
56    哥斯达黎加     8482     4999441   1697
57       挪威     9001     5314336   1694
58     萨尔瓦多    10303     6420744   1605
59     爱沙尼亚     2015     1320884   1525
60     巴勒斯坦     6764     4684777   1444
61       芬兰     7301     5518050   1323
62    阿尔巴尼亚     3667     2866376   1279
63       捷克    13341    10625695   1256
64    毛里塔尼亚     5518     4403319   1253
65      乌克兰    55931    44622516   1253
66     巴基斯坦   253604   212215030   1195
67      孟加拉   190057   161356039   1178
68     保加利亚     7645     7024216   1088
69     斯威士兰     1434     1390000   1032
70       波兰    38457    37978548   1013
71    几内亚比绍     1842     1874309    983
72     克罗地亚     3827     4089400    936
73      阿富汗    34740    37172386    935
74    中非共和国     4356     4666377    933
75    斯洛文尼亚     1859     2067372    899
76     塞浦路斯     1023     1189265    860
77       埃及    83930    98423595    853
78       加纳    24988    29767108    839
79    塔吉克斯坦     6643     9100837    730
80      立陶宛     1875     2789533    672
81       印度   906752  1352617328    670
82     拉脱维亚     1174     1926542    609
83      尼泊尔    17061    28087871    607
84       海地     6727    11123176    605
85      喀麦隆    15173    25216237    602
86      菲律宾    57545   106651922    540
87     科特迪瓦    13037    24900000    524
88     塞内加尔     8243    15854360    520
89      几内亚     6200    12414318    499
90     尼加拉瓜     3147     6465513    487
91    阿尔及利亚    20216    42228429    479
92      摩洛哥    16097    36029138    447
93      巴拉圭     3074     6956071    442
94      匈牙利     4258     9768785    436
95   乌兹别克斯坦    14085    32955400    427
96     澳大利亚    10487    24992369    420
97    刚果（布）     2028     5260000    386
98       希腊     3883    10727668    362
99   

+ 最后的rate表示的是 外国每百万人的确诊人数
+ 可以看到卡塔尔每百万人就有将近4万人确诊，差不多等于每100人中就有4个人确诊
+ 看似概率小，但其实如果不控制住疫情，这个概率会翻倍的往上升

## 分析美国历史每日数据

In [21]:
# 查看美国每日数据
usa_day_data = pd.read_csv(filepath_fg+"/部分国家历史每日数据/美国.csv")
# 列名中文化
usa_day_data.columns = ['日期', '新增确诊', '累计确诊', '累计治愈', '累计死亡']
usa_day_data

日期   新增确诊     累计确诊     累计治愈    累计死亡
0    1.28      0        5        0       0
1    1.29      0        5        0       0
2    1.30      1        6        0       0
3    1.31      0        6        0       0
4    2.01      1        7        0       0
5    2.02      1        8        0       0
6    2.03      3       11        1       0
7    2.04      0       11        1       0
8    2.05      0       11        1       0
9    2.06      1       12        1       0
10   2.07      0       12        1       0
11   2.08      0       12        1       0
12   2.09      0       12        1       0
13   2.10      0       12        1       0
14   2.11      1       13        1       0
15   2.12      0       13        3       0
16   2.13      1       14        3       0
17   2.14      1       15        3       0
18   2.15      0       15        3       0
19   2.16      0       15        3       0
20   2.17      0       15        3       0
21   2.18      0       15        3       0
22   2.19      0       15        3       0
23   2.20      0       15        3       0
24   2.21      0       15        3       0
25   2.22     19       34        3       0
26   2.23      0       34        3       0
27   2.24      1       35        3       0
28   2.25     18       53        3       0
29   2.26      4       57        3       0
30   2.27      3       60        3       0
31   2.28      0       60        3       0
32   2.29      4       64        3       0
33   3.01      9       69        3       1
34   3.02     20       89        3       2
35   3.03     16      105        3       6
36   3.04     21      126        3       9
37   3.05     35      161        8      11
38   3.06     73      234        8      14
39   3.07    103      337       10      17
40   3.08    118      455       10      19
41   3.09    127      582       10      22
42   3.10    173      755       10      26
43   3.11    297     1052       15      31
44   3.12    269     1329       15      38
45   3.13    551     1880       31      41
46   3.14    356     2236       31      49
47   3.15    697     2943       48      60
48   3.16    831     3774       73      69
49   3.17    913     4687       74      93
50   3.18   1835     6522      106     116
51   3.19   3737    10259      106     152
52   3.20   4106    14365      125     212
53   3.21   5811    20176      147     277
54   3.22   6680    26856      178     346
55   3.23   8368    35224      178     471
56   3.24  11219    46443      314     586
57   3.25   8799    55242      354     801
58   3.26  13981    69223      619    1054
59   3.27  16820    86043      753    1304
60   3.28  18794   104839      869    1716
61   3.29  19837   124676     1095    2196
62   3.30  17954   143071     4856    2513
63   3.31  22803   165874     5995    3178
64   4.01  24215   190089     7141    4102
65   4.02  27174   217263     8710    5151
66   4.03  28395   245658     9311    6069
67   4.04  32910   278568     9920    7163
68   4.05  33913   312481    15021    8503
69   4.06  25490   337971    17582    9654
70   4.07  31098   369069    20003   11018
71   4.08  30910   399979    22539   12912
72   4.09  32600   432579    24213   14830
73   4.10  34605   467184    26522   16736
74   4.11  34517   501701    29192   18781
75   4.12  29129   530830    32314   20646
76   4.13  28169   558999    42009   22154
77   4.14  24221   583220    44319   23654
78   4.15  26775   609995    50045   26069
79   4.16  29567   640014    52772   31002
80   4.17  32165   672293    56243   33325
81   4.18  28866   706856    60523   37087
82   4.19  26070   735366    66854   39095
83   4.20  28088   760570    71011   40702
84   4.21  25985   788920    73533   42458
85   4.22  29973   826248    75682   45153
86   4.23  31900   843981    77064   46859
87   4.24  38958   871285    80937   50066
88   4.25  35419   907096    99129   52063
89   4.26  26509   941628   105823   54024
90   4.27  23196   967585   107060   54931
91   4.28  25409   990135   111587   56475
92   4.2

+ 可以发现在1月和2月，美国只累计确诊不到一百例
+ 但是从三月开始，确诊病例逐渐增多，不到一个月就超过我国7个总共累计确诊病例数
+ 而且每日新增病例在三月底就是过万了，趋势仍在不断上升
+ 一直到7月份，美国的每日新增确诊病例已到达了6万5千多人
+ 照这个趋势下去，再过不久，美国的每日新增病例就能达到我国累计确诊病例数了
+ 这数据很好的彰显了美国抗疫的失败，我国一个月就能控制的住，他们五六个月都搞不定

## 分析美国具体行政区数据

In [22]:
# 查看美国具体行政区数据
usa_region_data = pd.read_csv(filepath_fg+"/部分国家具体行政区数据/美国-具体行政区数据.csv", encoding='GBK')
# 去除不要的列和列名中文化
del usa_region_data['date'] # 日期也不需要，因为都一样，在可视化时可以直接提出来
del usa_region_data['isUpdated']
del usa_region_data['confirmAdd']
del usa_region_data['confirmAddCut']
del usa_region_data['suspect']
usa_region_data.columns = ['地区', '地区英文名', '累计确诊', '累计死亡', '累计治愈']
usa_region_data

地区                 地区英文名    累计确诊   累计死亡    累计治愈
0        纽约              New York  413315  32520   58366
1     加利福尼亚            California  409305   7890  108711
2      佛罗里达               Florida  369834   5206     381
3      德克萨斯                 Texas  351113   4213  186529
4       新泽西            New Jersey  177256  15826    3973
5      伊利诺伊              Illinois  163689   7324   75880
6       乔治亚               Georgia  148988   3254     340
7      亚利桑那               Arizona  148683   2918     174
8      马萨诸塞         Massachusetts  114033   8450   94347
9     宾夕法尼亚          Pennsylvania  107414   7046    1369
10    北卡罗来纳        North Carolina  102871   1668   67124
11   路易斯安那州             Louisiana   96583   3608   53288
12     密歇根州              Michigan   83059   6382   55162
13      田纳西             Tennessee   81944    872   47853
14      马里兰              Maryland   79545   3402    5380
15     弗吉尼亚              Virginia   79371   2049   10107
16     俄亥俄州                  Ohio   77216   3220    7630
17   南卡罗来纳州        South Carolina   73101   1203    5076
18    阿拉巴马州               Alabama   70358   1303   29736
19     印第安纳               Indiana   57916   2846    3169
20     华盛顿州            Washington   48728   1461    5488
21     康涅狄格           Connecticut   48096   4406      63
22     明尼苏达             Minnesota   47457   1548   41511
23    密西西比州           Mississippi   45524   1389   30315
24     威斯康星             Wisconsin   44135    860   33902
25     科罗拉多              Colorado   41071   1763    3671
26     爱荷华州                  Iowa   39471    801   28371
27      内华达                Nevada   37605    681   25900
28     密苏里州              Missouri   37161   1183    4102
29       犹他                  Utah   35012    251    4006
30     阿肯色州              Arkansas   34655    374   27283
31  退伍军人事务部        veteran affair   33411   1927   13019
32       美军                   NaN   31418     48   12728
33    俄克拉荷马              Oklahoma   26326    461   20664
34      肯塔基              Kentucky   24060    674    6927
35    内布拉斯加              Nebraska   23173    317   14641
36     堪萨斯州                Kansas   22929    316   13095
37      罗德岛          Rhode Island   17986    996    1683
38    新墨西哥州            New Mexico   17517    588    6870
39     爱达荷州                 Idaho   15822    126    3022
40      俄勒冈                Oregon   15139    269    1885
41     特拉华州              Delaware   13746    525    7362
42     波多黎各           Puerto Rico   12940    180     850
43    华盛顿特区  District of Columbia   11427    580    1932
44    联邦监狱局                   NaN   10811     99    6195
45  纳瓦霍族保留地                   NaN    8617    425     930
46    南达科塔州          South Dakota    8019    118    7081
47    新罕布什尔         New Hampshire    6262    400    5316
48    北达科他州          North Dakota    5207     94    4319
49    西弗吉尼亚         West Virginia    5172    101    3491
50      缅因州                 Maine    3723    118    3191
51      蒙大拿               Montana    2712     40    1493
52     怀俄明州               Wyoming    2238     25    1694
53    阿拉斯加州                Alaska    2042     18     737
54      夏威夷                Hawaii    1418     24    1084
55      佛蒙特               Vermont    1366     56    1148
56       关岛                  Guam     319      5     143
57  美属维尔京群岛     US Virgin Islands     304      6     135
58  至尊公主号邮轮                   NaN     103      3       0
59  钻石公主号邮轮                   NaN      49      0       0
60  北马里亚纳群岛                   NaN      31      2      16
61     撤回侨民                   NaN       4      0       0
62    地区待确认                   NaN       0      0       0

In [23]:
usa_region_data.sort_values(by='累计确诊', ascending = False) # 通过累计确诊纬度分析

地区                 地区英文名    累计确诊   累计死亡    累计治愈
0        纽约              New York  413315  32520   58366
1     加利福尼亚            California  409305   7890  108711
2      佛罗里达               Florida  369834   5206     381
3      德克萨斯                 Texas  351113   4213  186529
4       新泽西            New Jersey  177256  15826    3973
5      伊利诺伊              Illinois  163689   7324   75880
6       乔治亚               Georgia  148988   3254     340
7      亚利桑那               Arizona  148683   2918     174
8      马萨诸塞         Massachusetts  114033   8450   94347
9     宾夕法尼亚          Pennsylvania  107414   7046    1369
10    北卡罗来纳        North Carolina  102871   1668   67124
11   路易斯安那州             Louisiana   96583   3608   53288
12     密歇根州              Michigan   83059   6382   55162
13      田纳西             Tennessee   81944    872   47853
14      马里兰              Maryland   79545   3402    5380
15     弗吉尼亚              Virginia   79371   2049   10107
16     俄亥俄州                  Ohio   77216   3220    7630
17   南卡罗来纳州        South Carolina   73101   1203    5076
18    阿拉巴马州               Alabama   70358   1303   29736
19     印第安纳               Indiana   57916   2846    3169
20     华盛顿州            Washington   48728   1461    5488
21     康涅狄格           Connecticut   48096   4406      63
22     明尼苏达             Minnesota   47457   1548   41511
23    密西西比州           Mississippi   45524   1389   30315
24     威斯康星             Wisconsin   44135    860   33902
25     科罗拉多              Colorado   41071   1763    3671
26     爱荷华州                  Iowa   39471    801   28371
27      内华达                Nevada   37605    681   25900
28     密苏里州              Missouri   37161   1183    4102
29       犹他                  Utah   35012    251    4006
30     阿肯色州              Arkansas   34655    374   27283
31  退伍军人事务部        veteran affair   33411   1927   13019
32       美军                   NaN   31418     48   12728
33    俄克拉荷马              Oklahoma   26326    461   20664
34      肯塔基              Kentucky   24060    674    6927
35    内布拉斯加              Nebraska   23173    317   14641
36     堪萨斯州                Kansas   22929    316   13095
37      罗德岛          Rhode Island   17986    996    1683
38    新墨西哥州            New Mexico   17517    588    6870
39     爱达荷州                 Idaho   15822    126    3022
40      俄勒冈                Oregon   15139    269    1885
41     特拉华州              Delaware   13746    525    7362
42     波多黎各           Puerto Rico   12940    180     850
43    华盛顿特区  District of Columbia   11427    580    1932
44    联邦监狱局                   NaN   10811     99    6195
45  纳瓦霍族保留地                   NaN    8617    425     930
46    南达科塔州          South Dakota    8019    118    7081
47    新罕布什尔         New Hampshire    6262    400    5316
48    北达科他州          North Dakota    5207     94    4319
49    西弗吉尼亚         West Virginia    5172    101    3491
50      缅因州                 Maine    3723    118    3191
51      蒙大拿               Montana    2712     40    1493
52     怀俄明州               Wyoming    2238     25    1694
53    阿拉斯加州                Alaska    2042     18     737
54      夏威夷                Hawaii    1418     24    1084
55      佛蒙特               Vermont    1366     56    1148
56       关岛                  Guam     319      5     143
57  美属维尔京群岛     US Virgin Islands     304      6     135
58  至尊公主号邮轮                   NaN     103      3       0
59  钻石公主号邮轮                   NaN      49      0       0
60  北马里亚纳群岛                   NaN      31      2      16
61     撤回侨民                   NaN       4      0       0
62    地区待确认                   NaN       0      0       0

+ 首先可以看到，美国很多地区的累计确诊人数都超过了我国全国的累计确诊人数
+ 其次可以看到一些异常点，佛罗丽达的确诊人数有30万，但是累计治愈才381人，同时他们的死亡人数为5206（截至数据爬取当日）
+ 这就很迷，这可能有大多数确诊病例没有去治疗或者没有统计，毕竟美国政府一直在阻拦抗疫
+ 这与我国就形成了鲜明的对比，我国是尽全力的医治每一个感染者
+ 而美国是，只要不统计就是没有确诊，很美国啊

# 其他的数据分析可通过数据可视化直观看出来